src/core/plugin/luapi_application.h

```diff
@@ -327,6 +384,263 @@ static int applib_layerAction(lua_State* L) {
     return 1;
 }
 
+static int addStrokeHelper(lua_State* L, Stroke* stroke, Layer* layer) {
+    //    Plugin* plugin = Plugin::getPluginFromLua(L);
+    //    Control* ctrl = plugin->getControl();
+    //    PageRef const& page = ctrl->getCurrentPage();
+    //    Layer* layer = page->getSelectedLayer();
+    //    Stroke* stroke = new Stroke();
+
+    // discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TTABLE);
+
+    // Make sure we have enough points to form a stroke
+    if (stroke->getPointCount() >= 2) {
+        // Get the rest of the attributes. Default to something if none is provided
+        // TODO: (willnilges) Test this. Is this robust? Should we be doing this one field at a time?
+        lua_getfield(L, 1, "width");
+        lua_getfield(L, 1, "color");
+        lua_getfield(L, 1, "fill");
+        lua_getfield(L, 1, "tool");
+        lua_getfield(L, 1, "lineStyle");
+        double width = luaL_optnumber(L, -5, 1.5);
+        long long int color = luaL_optinteger(L, -4, 0);
+        int fill = luaL_optinteger(L, -3, -1);
+        const char* tool = luaL_optstring(L, -2, "pen");
+        const char* style = luaL_optstring(L, -1, "");
+        lua_pop(L, 5);
+
+        stroke->setWidth(width);
+        stroke->setColor(Color(color));
+        stroke->setFill(fill);
+        if (strcmp("eraser", tool) == 0) {
+            stroke->setToolType(STROKE_TOOL_ERASER);
+        } else if (strcmp("pen", tool) == 0) {
+            stroke->setToolType(STROKE_TOOL_PEN);
+        } else if (strcmp("highlighter", tool) == 0) {
+            stroke->setToolType(STROKE_TOOL_HIGHLIGHTER);
+        } else {
+            g_warning("%s", FC(_F("Unknown stroke type: \"{1}\", assuming pen") % tool));
+        }
+        LineStyle lineStyle = StrokeStyle::parseStyle(style);
+        stroke->setLineStyle(lineStyle);
+        if (strcmp("", style) != 0 && strcmp("dash", style) != 0 && strcmp("dashdot", style) != 0 &&
+            strcmp("dot", style) != 0)
+            g_warning("%s", FC(_F("Unknown style: \"{1}\", assuming solid") % style));
+        layer->addElement(stroke);
+        stroke = nullptr;
+        return 1;
+    }
+    // If there aren't at least 2 points, then don't add the stroke.
+    g_warning("Stroke shorter than two points. Discarding. (Has %d/2)", stroke->getPointCount());
+    delete stroke;
+    stroke = nullptr;
+    return 1;
+}
+
+/**
+ * Given a table containing a series of splines, draws a stroke on the canvas.
+ * Expects a table of coordinate pairs along with attributes of the stroke.
+ * Attributes are optional.
```

Comment:
Going to add this to the documentation.
```
The function expects 8 points per spline. Due to the nature of quadratic                                                                                
splines, you must pass your points in a repeating pattern:
startX, startY, ctrl1X, ctrl1Y, ctrl2X, ctrl2Y, endX, endY, startX, startY, ...

The function checks that the spline table is divisible by eight, and will throw
an error if it is not.       
```

Reply:
> Re: how to pass this data in the first place I've thought about it some more, and what if we made a struct? One probably already exists in the Spline PR, right? It would be annoying, but not difficult to pass a table of tables. That might be cleaner and perhaps easier to understand for a first time user. It would also make it more logical to add cubic splines and the like. I don't have any problem with the current approach, but it's not super extensible, and is entirely dependent on the interpretation done by the function.

I have thought about it, too. The downside of passing a table of tables (consisting of 8 coordinates each) is, that it takes more computation time to check, whether all individual tables are valid. Moreover it will also take more memory to store/pass it like that. So probably it is better to keep it as one single table consisting of coordinates. For the user it is not that a big difference, I think. Efficiency is more important.